app/models/spree/gateway/amazon.rb

```diff
@@ -112,7 +112,13 @@ def authorize(amount, amazon_checkout, gateway_options={})
           message = "#{mws_res.code} #{mws_res.body}"
         end
       end
-
+      order.amazon_transaction.update!(
+        authorization_success: success,
+        message: message,
+        authorization_reference_id: authorization_reference_id,
+        soft_decline: amazon_response.soft_decline?,
+        retry: !success
+      )
```

Comment:
This method has grown considerably. Maybe at least add a comment above saying why you're doing that.

Comments are generally greatly replaced by method names .. which speak for themselves.  Ideally maybe the whole method should go into a class of its own, as adding helper methods in this class will only intermix them with the other gateway methods.